src/server/verification/verificationAPI.js

```diff
@@ -325,30 +342,38 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
     onlyInEnv('production', 'staging', 'test'),
     wrapAsync(async (req, res, next) => {
       const log = req.log.child({ from: 'verificationAPI - verify/sendemail' })
+
       const { user, body } = req
-      // log.info({ user, body })
+      const { email } = body.user
+
       //merge user details for use by mautic
       let userRec: UserRecord = _.defaults(body.user, user)
+
       if (conf.allowDuplicateUserData === false && (await storage.isDupUserData(userRec))) {
         return res.json({ ok: 0, error: 'Email already exists, please use a different one' })
       }
+
       if (!user.mauticId) {
         //first time create contact for user in mautic
         const mauticContact = await Mautic.createContact(userRec)
         userRec.mauticId = mauticContact.contact.fields.all.id
         log.debug('created new user mautic contact', userRec)
       }
+
       if (conf.skipEmailVerification === false) {
         const code = generateOTP(6)
+
         if (!user.isEmailConfirmed) {
           Mautic.sendVerificationEmail(userRec, code)
           log.debug('send new user email validation code', code)
         }
+
         // updates/adds user with the emailVerificationCode to be used for verification later and with mauticId
         storage.updateUser({
           identifier: user.loggedInAs,
           mauticId: userRec.mauticId,
-          emailVerificationCode: code
+          emailVerificationCode: code,
+          otp: { email }
```

Comment:
wouldn't this overwrite the current otp record values?

Reply:
@sirpy, no, it doesn't. The gun DB is merging incoming data, not replacing the existing objects with the new one.

Reply:
its not gundb is mongodb used for the private user database

Reply:
@sirpy done. see new commit